bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/process/StreamListener.java

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ * Copyright 2023 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.debug.gdbjtag.openocd.dsf.process;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+
+import org.eclipse.debug.core.IStreamListener;
+import org.eclipse.debug.core.model.IFlushableStreamMonitor;
+import org.eclipse.debug.core.model.IProcess;
+import org.eclipse.debug.core.model.IStreamMonitor;
+import org.eclipse.ui.console.IOConsoleOutputStream;
+
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.debug.gdbjtag.openocd.dsf.console.IdfProcessConsole;
+import com.espressif.idf.debug.gdbjtag.openocd.dsf.console.IdfProcessConsoleFactory;
+
+/**
+ * This class listens to a specified stream monitor to get notified on output from the process connected to console.
+ * Its separated so that we can apply filters and pass the text onto the respected stream.
+ * It is designed to handle both error and standard stream
+ * @author Ali Azam Rana
+ */
+public class StreamListener implements IStreamListener
```

Comment:
If this is a derived class, could you mention the original class details and what part of the code is modified in that for the future code maintenance point of view?

Reply:
done